src/weights.jl

```diff
@@ -1,19 +1,32 @@
 
 ###### Weight vector #####
 
-immutable WeightVec{W,Vec<:RealVector}
-    values::Vec
-    sum::W
+if VERSION < v"0.6.0-dev.2123"
+    immutable WeightVec{S<:Real, T<:Real, V<:RealVector} <: RealVector{T}
+        values::V
+        sum::S
+    end
+else
+    immutable WeightVec{S<:Real, T<:Real, V<:AbstractVector{T}} <: AbstractVector{T}
+        values::V
+        sum::S
+    end
 end
 
 """
-    WeightVec(vs, [wsum])
+    function WeightVec{S<:Real, V<:RealVector}(vs::V, s::S)
```

Comment:
This change is incorrect, the signature should just reflect how the user can call the function, not how it's implemented.